lib/spack/spack/filesystem_view.py

```diff
@@ -201,9 +201,12 @@ def __init__(self, root, layout, **kwargs):
                 # Write projections file to new view
                 # Not strictly necessary as the empty file is the empty
                 # projection but it makes sense for consistency
-                mkdirp(os.path.dirname(projections_path))
-                with open(projections_path, 'w') as f:
-                    f.write(s_yaml.dump({'projections': self.projections}))
+                try:
+                    mkdirp(os.path.dirname(projections_path))
+                    with open(projections_path, 'w') as f:
+                        f.write(s_yaml.dump({'projections': self.projections}))
+                except OSError:
+                    pass
```

Comment:
This should raise when `self.projections` is non-empty.

Maybe we should do without the file entirely when `self.projections` is empty, but that could be a separate PR optimization as well.

Since extensions are activated using views with the default projections, raising `if self.projections` should not effect the bugfix in this PR.

```
except OSError as e:
    if self.projections:
        raise e
```

Reply:
Done. I confirmed that I am still able to uninstall extensions with this change. Other than that, I really don't know much about this file.